zstd_stream.go

```diff
@@ -146,6 +147,44 @@ func (w *Writer) Close() error {
 	return nil
 }
 
+// cSize is the recommended size of reader.compressionBuffer. This func and
+// invocation allow for a one-time check for validity.
+var cSize = func() int {
+	v := int(C.ZBUFF_recommendedDInSize())
+	if v <= 0 {
+		panic(fmt.Errorf("ZBUFF_recommendedDInSize() returned invalid size: %v", v))
+	}
+	return v
+}()
+
+// dSize is the recommended size of reader.decompressionBuffer. This func and
+// invocation allow for a one-time check for validity.
+var dSize = func() int {
+	v := int(C.ZBUFF_recommendedDOutSize())
+	if v <= 0 {
+		panic(fmt.Errorf("ZBUFF_recommendedDOutSize() returned invalid size: %v", v))
+	}
+	return v
+}()
+
+// cPool is a pool of buffers for use in reader.compressionBuffer. Buffers are
+// taken from the pool in NewReaderDict, returned in reader.Close()
+var cPool = sync.Pool{
+	New: func() interface{} {
+		buff := make([]byte, cSize)
+		return &buff
+	},
+}
+
+// dPool is a pool of buffers for use in reader.decompressionBuffer. Buffers are
+// taken from the pool in NewReaderDict, returned in reader.Close()
+var dPool = sync.Pool{
+	New: func() interface{} {
+		buff := make([]byte, dSize)
```

Comment:
I wonder if this should have a comment like "return *[]byte since sync.Pool documentation recommends returning pointer types to avoid extra allocations"